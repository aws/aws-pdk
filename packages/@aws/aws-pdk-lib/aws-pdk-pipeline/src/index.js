"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./pdk-pipeline"), exports);
__exportStar(require("./pdk-pipeline-ts-project"), exports);
__exportStar(require("./pdk-pipeline-py-project"), exports);
__exportStar(require("./pdk-pipeline-java-project"), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7QUFBQSxpREFBK0I7QUFDL0IsNERBQTBDO0FBQzFDLDREQUEwQztBQUMxQyw4REFBNEMiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmVcIjtcbmV4cG9ydCAqIGZyb20gXCIuL3Bkay1waXBlbGluZS10cy1wcm9qZWN0XCI7XG5leHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmUtcHktcHJvamVjdFwiO1xuZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lLWphdmEtcHJvamVjdFwiO1xuIl19