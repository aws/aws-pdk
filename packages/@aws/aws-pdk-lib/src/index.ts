export * as construct from "./construct";
export * as project from "./project";