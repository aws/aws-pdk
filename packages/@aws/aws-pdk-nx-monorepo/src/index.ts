export * as project from "./project";
