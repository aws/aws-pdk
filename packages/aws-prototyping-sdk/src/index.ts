// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export class Hello {
  public sayHello() {
    return "hello, world!";
  }
}
