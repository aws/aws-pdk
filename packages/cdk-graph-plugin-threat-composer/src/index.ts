/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./plugin";
export * from "./model-generator/types";
