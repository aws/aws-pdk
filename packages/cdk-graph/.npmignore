# ~~ Generated by projen. To modify, edit .projenrc.js and run "pnpm exec projen".
/.projen/
/test-reports/
junit.xml
/coverage/
/.prettierignore
/.prettierrc.json
/test/
/tsconfig.dev.json
/src/
!/lib/
!/lib/**/*.js
!/lib/**/*.d.ts
dist
/tsconfig.json
/.github/
/.vscode/
/.idea/
/.projenrc.js
tsconfig.tsbuildinfo
/.eslintrc.json
!.jsii
/.gitattributes
