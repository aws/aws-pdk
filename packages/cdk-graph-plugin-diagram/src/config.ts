/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { FilterPreset, IGraphFilterPlan } from "@aws/cdk-graph";
import { GraphThemeConfigProp } from "./internal/graphviz";

/**
 * Supported diagram formats that can be generated.
 *
 * Extended formats are automatically generated, for example if you generate "png" which extends
 * "svg" which extends "dot", the resulting generated files will be all aforementioned.
 */
export enum DiagramFormat {
  /**
   * Graphviz [DOT Language](https://graphviz.org/doc/info/lang.html)
   *
   * @default
   */
  DOT = "dot",
  /**
   * [SVG](https://developer.mozilla.org/en-US/docs/Web/SVG) generated
   * using [dot-wasm](https://hpcc-systems.github.io/hpcc-js-wasm/classes/graphviz.Graphviz.html)
   * from {@link DiagramFormat.DOT} file.
   *
   * @extends {DiagramFormat.DOT}
   */
  SVG = "svg",
  /**
   * [PNG](https://en.wikipedia.org/wiki/Portable_Network_Graphics) generated
   * using [sharp](https://sharp.pixelplumbing.com/api-output#png) from {@link DiagramFormat.SVG} file
   *
   * @extends {DiagramFormat.SVG}
   */
  PNG = "png",

  // TODO: add drawio support
}

/**
 * Positional coordinates for a node in inches
 */
export interface INodePosition {
  readonly x: number;
  readonly y: number;
}

/**
 * Base config to specific a unique diagram to be generated.
 * @struct
 */
export interface IDiagramConfigBase {
  /**
   * The output format(s) to generated.
   * @default `DiagramFormat.PNG` - which will through extension also generate `DiagramFormat.SVG` and `DiagramFormat.DOT`
   */
  readonly format?: DiagramFormat | DiagramFormat[];

  /** Graph {@link IGraphFilterPlan Filter Plan}  used to generate a unique diagram */
  readonly filterPlan?: IGraphFilterPlan;

  /** Config for graph theme */
  readonly theme?: GraphThemeConfigProp;

  /** Config for predetermined node positions given their CDK construct id. */
  readonly nodePositions?: { [cdkConstructId: string]: INodePosition };
}

/** Diagram configuration definition
 * @struct
 */
export interface IDiagramConfig extends IDiagramConfigBase {
  /**
   * Name of the diagram. Used as the basename of the generated file(s) which gets the extension appended.
   */
  readonly name: string;

  /**
   * The title of the diagram.
   */
  readonly title: string;

  /**
   * Indicates if default diagram config is applied as defaults to this config.
   *
   * @default false
   */
  readonly ignoreDefaults?: boolean;
}

/**
 * Plugin configuration for diagram plugin.
 * @struct
 */
export interface IPluginConfig {
  /** Default configuration to apply to all diagrams */
  readonly defaults?: IDiagramConfigBase;
  /** List of diagram configurations to generate diagrams */
  readonly diagrams?: IDiagramConfig[];
}

/** Default diagram title */
export const DEFAULT_DIAGRAM_TITLE = "Cloud Diagram";

/** Default diagram name */
export const DEFAULT_DIAGRAM_NAME = "diagram";

/** Default configuration settings */
export const CONFIG_DEFAULTS: IDiagramConfigBase = {
  format: [DiagramFormat.PNG],
  filterPlan: {
    preset: FilterPreset.COMPACT,
  },
};

/** Default diagram config */
export const DEFAULT_DIAGRAM: IDiagramConfig = {
  name: DEFAULT_DIAGRAM_NAME,
  title: DEFAULT_DIAGRAM_TITLE,
};
