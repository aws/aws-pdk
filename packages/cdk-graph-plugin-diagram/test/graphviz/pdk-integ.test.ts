/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { CdkGraph } from "@aws/cdk-graph";
import { PDKPipelineIntegApp } from "@aws/cdk-graph/test/__fixtures__/pdk-integ";
import * as fs from "fs-extra";
import * as testUtils from "./test-utils";
import { CdkGraphDiagramPlugin } from "../../src";

jest.setTimeout(90000); // CI tests timeout occasionally so increase to large timeout buffer

const makeCdkOutdir = async (name: string) =>
  testUtils.makeCdkOutDir("pdk-integ", name);

describe("pdk-integ", () => {
  describe("pipeline", () => {
    let outdir: string;
    let app: PDKPipelineIntegApp;
    let graph: CdkGraph;
    let plugin: CdkGraphDiagramPlugin;

    beforeAll(async () => {
      outdir = await makeCdkOutdir("pipeline");

      app = new PDKPipelineIntegApp({ outdir });
      plugin = new CdkGraphDiagramPlugin();
      graph = new CdkGraph(app, {
        plugins: [plugin],
      });
      app.synth();
      await graph.report();
    });

    it("should generate dot artifact", async () => {
      expect(plugin.defaultDotArtifact).toBeDefined();
      expect(
        await fs.pathExists(plugin.defaultDotArtifact!.filepath)
      ).toBeTruthy();
      expect(
        testUtils.cleanseDotSnapshot(
          await fs.readFile(plugin.defaultDotArtifact!.filepath, {
            encoding: "utf-8",
          })
        )
      ).toMatchSnapshot();
    });

    it("should generate png artifact", async () => {
      expect(plugin.defaultPngArtifact).toBeDefined();
      expect(
        await fs.pathExists(plugin.defaultPngArtifact!.filepath)
      ).toBeTruthy();

      await testUtils.expectToMatchImageSnapshot(
        plugin.defaultPngArtifact!.filepath,
        "pdk-integ-pipeline"
      );
    });
  });
});
