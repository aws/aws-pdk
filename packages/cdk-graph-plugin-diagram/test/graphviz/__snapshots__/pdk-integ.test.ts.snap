// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdk-integ pipeline should generate dot artifact 1`] = `
"digraph "ROOT" {
  label = "Cloud Diagram";
  imagepath = "<aws_arch_asset_directory";
  fontname = "Roboto Condensed";
  class = "font_regular";
  center = true;
  compound = true;
  concentrate = true;
  dpi = 300;
  fontcolor = "#000000";
  fontnames = "ps";
  fontsize = 14;
  forcelabels = true;
  labelloc = "tc";
  nodesep = 0.8;
  pad = "%2,%1";
  rankdir = "TB";
  ranksep = 0.75;
  ratio = "compress";
  remincross = true;
  size = "%1024,%1024!";
  splines = "ortho";
  stylesheet = "https://fonts.googleapis.com/css2?family=Roboto+Condensed:wght@300;400;700&display=swap";
  bgcolor = "#FFFFFF";
  edge [
    fontname = "Roboto Condensed";
    class = "font_light_italic";
    dir = "both";
    color = "#5A6B861A";
    penwidth = 0.75;
    arrowhead = "none";
    arrowtail = "normal";
    arrowsize = 0.5;
    fontsize = 9;
    style = "solid";
    fontcolor = "#5A6B861A";
  ];
  node [
    fontname = "Roboto Condensed";
    class = "font_regular";
    shape = "box";
    style = "solid";
    fixedsize = false;
    width = 0.25;
    height = 0.25;
    labelloc = "c";
    imagescale = true;
    fontsize = 10;
    penwidth = 0;
    fontcolor = "#232F3E";
  ];
  subgraph "cluster_stage_Prod" {
    fontname = "Roboto Condensed";
    class = "font_bold_italic";
    labelloc = "tc";
    fontsize = 12;
    style = "dashed";
    label = "Prod";
    margin = 6;
    rank = "same";
    color = "#B0084D";
    bgcolor = "transparent";
    fillcolor = "transparent";
    pencolor = "#B0084D";
    fontcolor = "#B0084D";
    subgraph "cluster_stack_ProdFixtureStack" {
      fontname = "Roboto Condensed";
      class = "font_light";
      labelloc = "tc";
      fontsize = 12;
      style = "solid,bold,filled";
      label = "FixtureStack";
      fillcolor = "#5A6B861A";
      margin = 10;
      color = "#000000CC";
      bgcolor = "#5A6B861A";
      pencolor = "transparent";
      fontcolor = "#000000CC";
      subgraph "cluster_ProdFixtureStackNetworkingLayerB7E3ED99" {
        fontname = "Roboto Condensed";
        class = "font_light";
        labelloc = "tc";
        fontsize = 12;
        style = "rounded,dashed";
        label = "NetworkingLayer";
        "node_ProdFixtureStackNetworkingLayerVPC389D72E2" [
          label = "VPC";
          comment = "nodeType:RESOURCE(AWS::EC2::VPC)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "networking_content_delivery/vpc/service_icon.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackNetworkingLayerWebSecurityGroup812D488E" [
          label = "WebSecurity\\nGroup";
          comment = "nodeType:RESOURCE(AWS::EC2::SecurityGroup)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.46;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/ec2/service_icon.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackNetworkingLayerwebserverroleF6CF2B99" [
          label = "WebserverRole";
          comment = "nodeType:RESOURCE(AWS::IAM::Role)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "security_identity_compliance/iam/role.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackNetworkingLayerGetUserDataParam348629C3" [
          label = "GetUserData\\nParam";
          comment = "nodeType:DEFAULT";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.46;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "management_governance/cloudformation/service_icon.svg";
        ];
        "node_ProdFixtureStackNetworkingLayerWebServer25D75DC4" [
          label = "WebServer";
          comment = "nodeType:RESOURCE(AWS::EC2::Instance)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/ec2/instance/c5.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackNetworkingLayerBastion5F69250B" [
          label = "Bastion";
          comment = "nodeType:RESOURCE(AWS::EC2::Instance)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/ec2/instance/t2.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
      subgraph "cluster_ProdFixtureStackDataLayerF4026588" {
        fontname = "Roboto Condensed";
        class = "font_light";
        labelloc = "tc";
        fontsize = 12;
        style = "rounded,dashed";
        label = "DataLayer";
        "node_ProdFixtureStackDataLayerBucketD02C835B" [
          label = "Bucket";
          comment = "nodeType:RESOURCE(AWS::S3::Bucket)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "storage/s3/bucket.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackDataLayerDatabase69D2A18D" [
          label = "Database";
          comment = "nodeType:RESOURCE(AWS::RDS::DBInstance)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "database/rds/instance/postgresql.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackDataLayerReadRoleD4070F87" [
          label = "ReadRole";
          comment = "nodeType:RESOURCE(AWS::IAM::Role)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "security_identity_compliance/iam/role.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
      subgraph "cluster_ProdFixtureStackApiLayer6B6FBC7E" {
        fontname = "Roboto Condensed";
        class = "font_light";
        labelloc = "tc";
        fontsize = 12;
        style = "rounded,dashed";
        label = "ApiLayer";
        "node_ProdFixtureStackApiLayerApiGatewayFD5A946B" [
          label = "ApiGateway";
          comment = "nodeType:RESOURCE(AWS::ApiGateway::RestApi)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "application_integration/api_gateway/service_icon.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackApiLayerHelloHandler72BEB90E" [
          label = "HelloHandler";
          comment = "nodeType:RESOURCE(AWS::Lambda::Function)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/lambda/lambda_function.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackApiLayerWorldHandler82F3FC55" [
          label = "WorldHandler";
          comment = "nodeType:RESOURCE(AWS::Lambda::Function)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/lambda/lambda_function.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
      subgraph "cluster_ProdFixtureStackWebsite71FAC1DE" {
        fontname = "Roboto Condensed";
        class = "font_light";
        labelloc = "tc";
        fontsize = 12;
        style = "rounded,dashed";
        label = "Website";
        "node_ProdFixtureStackWebsiteSourceBucketBF5786EE" [
          label = "SourceBucket";
          comment = "nodeType:RESOURCE(AWS::S3::Bucket)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "storage/s3/bucket.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackWebsiteConfigProvider228954C9" [
          label = "ConfigProvider";
          comment = "nodeType:DEFAULT";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "management_governance/cloudformation/service_icon.svg";
        ];
        "node_ProdFixtureStackWebsiteWebDistroE3AC5570" [
          label = "WebDistro";
          comment = "nodeType:RESOURCE(AWS::CloudFront::Distribution)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "networking_content_delivery/cloudfront/service_icon.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
      subgraph "cluster_ProdFixtureStackEdgeCases1F90BDF7" {
        fontname = "Roboto Condensed";
        class = "font_light";
        labelloc = "tc";
        fontsize = 12;
        style = "rounded,dashed";
        label = "EdgeCases";
        "node_ProdFixtureStackEdgeCasesLambdaWithAliasA9B3E2E7" [
          label = "LambdaWith\\nAlias";
          comment = "nodeType:RESOURCE(AWS::Lambda::Function)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.46;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/lambda/lambda_function.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdFixtureStackEdgeCasesImportedBucket00394349" [
          label = "ImportedBucket";
          comment = "nodeType:CFN_RESOURCE(AWS::S3::Bucket)";
          shape = "box";
          style = "filled,dotted";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "#5A6B861A";
          image = "storage/s3/bucket.svg";
          color = "#5A6B8633";
          fontcolor = "#5A6B86";
          penwidth = 1;
        ];
        "node_ProdFixtureStackEdgeCasesImportedLambda03BE5B8C" [
          label = "ImportedLambda";
          comment = "nodeType:CFN_RESOURCE(AWS::Lambda::Function)";
          shape = "box";
          style = "filled,dotted";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "#5A6B861A";
          image = "compute/lambda/lambda_function.svg";
          color = "#5A6B8633";
          fontcolor = "#5A6B86";
          penwidth = 1;
        ];
        "node_ProdFixtureStackEdgeCasesImportedResourceRoleE421C014" [
          label = "Imported\\nResourceRole";
          comment = "nodeType:RESOURCE(AWS::IAM::Role)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.46;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "security_identity_compliance/iam/role.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
    }
    subgraph "cluster_stack_ProdDependentStack" {
      fontname = "Roboto Condensed";
      class = "font_light";
      labelloc = "tc";
      fontsize = 12;
      style = "solid,bold,filled";
      label = "DependentStack";
      fillcolor = "#5A6B861A";
      margin = 10;
      color = "#000000CC";
      bgcolor = "#5A6B861A";
      pencolor = "transparent";
      fontcolor = "#000000CC";
      subgraph "cluster_nstack_ProdDependentStackResourceStack28CAA7DE" {
        fontname = "Roboto Condensed";
        class = "font_light_italic";
        labelloc = "tc";
        fontsize = 12;
        style = "solid,filled";
        label = "ResourceStack";
        fillcolor = "#5A6B861A";
        margin = 6;
        color = "#000000CC";
        bgcolor = "#5A6B861A";
        pencolor = "transparent";
        fontcolor = "#000000CC";
        "node_ProdDependentStackResourceStackLambdaFunction68E462DF" [
          label = "LambdaFunction";
          comment = "nodeType:RESOURCE(AWS::Lambda::Function)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "compute/lambda/lambda_function.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
        "node_ProdDependentStackResourceStackRole2DED6E96" [
          label = "Role";
          comment = "nodeType:RESOURCE(AWS::IAM::Role)";
          shape = "box";
          style = "solid,rounded";
          fixedsize = true;
          width = 1;
          height = 1.23;
          labelloc = "b";
          imagescale = true;
          imagepos = "tc";
          fillcolor = "transparent";
          image = "security_identity_compliance/iam/role.svg";
          color = "#232F3E";
          fontcolor = "#232F3E";
        ];
      }
    }
  }
  "node_ProdFixtureStackNetworkingLayerWebSecurityGroup812D488E" -> "node_ProdFixtureStackNetworkingLayerVPC389D72E2" [
    id = "REF:ooYEO";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackNetworkingLayerWebServer25D75DC4" -> "node_ProdFixtureStackNetworkingLayerwebserverroleF6CF2B99" [
    id = "REF:Z1FEBCC";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackNetworkingLayerWebServer25D75DC4" -> "node_ProdFixtureStackNetworkingLayerWebSecurityGroup812D488E" [
    id = "ATT:ZS3Naj";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackNetworkingLayerWebServer25D75DC4" -> "node_ProdFixtureStackNetworkingLayerVPC389D72E2" [
    id = "REF:2pWFWa";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackNetworkingLayerWebServer25D75DC4" -> "node_ProdFixtureStackNetworkingLayerGetUserDataParam348629C3" [
    id = "ATT:Uks1G";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackNetworkingLayerBastion5F69250B" -> "node_ProdFixtureStackNetworkingLayerVPC389D72E2" [
    id = "REF:22tne5";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackDataLayerDatabase69D2A18D" -> "node_ProdFixtureStackNetworkingLayerVPC389D72E2" [
    id = "REF:Z1nEbXt";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackDataLayerReadRoleD4070F87" -> "node_ProdFixtureStackDataLayerBucketD02C835B" [
    id = "ATT:1NySn2";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackDataLayerReadRoleD4070F87" -> "node_ProdFixtureStackDataLayerDatabase69D2A18D" [
    id = "REF:14mncW";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackApiLayerApiGatewayFD5A946B" -> "node_ProdFixtureStackApiLayerHelloHandler72BEB90E" [
    id = "ATT:J0Nar";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackApiLayerApiGatewayFD5A946B" -> "node_ProdFixtureStackApiLayerWorldHandler82F3FC55" [
    id = "ATT:25pby2";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackApiLayerHelloHandler72BEB90E" -> "node_ProdFixtureStackDataLayerBucketD02C835B" [
    id = "REF:Z20ifLV";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackApiLayerHelloHandler72BEB90E" -> "node_ProdFixtureStackDataLayerDatabase69D2A18D" [
    id = "REF:pH9Vu";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackWebsiteConfigProvider228954C9" -> "node_ProdFixtureStackWebsiteSourceBucketBF5786EE" [
    id = "ATT:r2Fpf";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackWebsiteConfigProvider228954C9" -> "node_ProdFixtureStackApiLayerApiGatewayFD5A946B" [
    id = "REF:Z2ng3O5";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackWebsiteWebDistroE3AC5570" -> "node_ProdFixtureStackWebsiteSourceBucketBF5786EE" [
    id = "ATT:Z18WhNR";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackEdgeCasesImportedResourceRoleE421C014" -> "node_ProdFixtureStackEdgeCasesImportedBucket00394349" [
    id = "IMP:PzhSU";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdFixtureStackEdgeCasesImportedResourceRoleE421C014" -> "node_ProdFixtureStackEdgeCasesImportedLambda03BE5B8C" [
    id = "IMP:ZeSpD5";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdDependentStackResourceStackRole2DED6E96" -> "node_ProdDependentStackResourceStackLambdaFunction68E462DF" [
    id = "ATT:jkhLR";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
  "node_ProdDependentStackResourceStackLambdaFunction68E462DF" -> "node_ProdDependentStackResourceStackRole2DED6E96" [
    ltail = "cluster_nstack_ProdDependentStackResourceStack28CAA7DE";
    id = "REF:ZKUqqz";
    penwidth = 0.75;
    arrowhead = "normal";
    arrowtail = "none";
    arrowsize = 0.75;
    style = "solid";
    color = "#5A6B86";
    fontcolor = "#5A6B86";
  ];
}"
`;
